enterprise/backend/src/metabase_enterprise/sso/integrations/jwt.clj

```diff
@@ -93,22 +94,38 @@
           user         (fetch-or-create-user! first-name last-name email login-attrs)
           session      (api.session/create-session! :sso user (req.util/device-info request))]
       (sync-groups! user jwt-data)
-      (mw.session/set-session-cookies request (response/redirect redirect-url) session (t/zoned-date-time (t/zone-id "GMT"))))))
+      {:session session, :redirect-url redirect-url, :jwt-data jwt-data})))
 
 (defn- check-jwt-enabled []
   (api/check (sso-settings/jwt-enabled)
     [400 (tru "JWT SSO has not been enabled and/or configured")]))
 
+(defn ^:private generate-response-token
+  [session jwt-data]
+  (validation/check-embedding-enabled)
+  (response/response {:id  (:id session)
+                      :exp (:exp jwt-data)
+                      :iat (:iat jwt-data)}))
+
+(defn ^:private redirect-to-idp
+  [idp redirect]
+  (let [return-to-param (if (str/includes? idp "?") "&return_to=" "?return_to=")]
+    (response/redirect (str idp (when redirect
+                                  (str return-to-param redirect))))))
+
+(defn ^:private handle-jwt-authentication
+  [{:keys [session redirect-url jwt-data]} token request]
+  (if token
+    (generate-response-token session jwt-data)
+    (mw.session/set-session-cookies request (response/redirect redirect-url) session (t/zoned-date-time (t/zone-id "GMT")))))
+
 (defmethod sso.i/sso-get :jwt
-  [{{:keys [jwt redirect]} :params, :as request}]
+  [{{:keys [jwt redirect token] :or {token false}} :params, :as request}]
```

Comment:
I'd probably let this be `nil` when not provided. It'll behave the same as `false` in conditionals, just a bit more idiomatic.
```suggestion
  [{{:keys [jwt redirect token]} :params, :as request}]
```

Reply:
In general, I should default to `nil` instead of `false` for these types of things?

Reply:
`nil` means you don't have it, `false` means you have it but it's, uh, false. Not being provided in a map is a classic nil case.

Reply:
Ok, perfect. Thanks for the tip